src/EGraphs/unionfind.jl

```diff
@@ -1,9 +1,43 @@
+"""
+A unionfind data structure is an eventually-idempotent endomorphism on `1:n`.
+
+What does this mean? It is a function `parents : 1:n -> 1:n` such that
+
+```
+parents^(n) = parents^(n+1)
+```
+
+This means that if you start at any `i ∈ 1:n`, repeatedly apply `parents` to `i`
+will eventually reach a fixed-point. We call the fixed points of `parents` the "roots" of the unionfind. This is implemented in [`find`].
+
+For the remaining discussion, let `parents*` the function which sends `i ∈ 1:n` to its fixed point with respect to `parents`.
+
+The point of a unionfind is to store a *partition* of `1:n`. To test if two
+elements `i, j ∈ 1:n` are in the same partition, we check if
+
+```
+parents*(i) = parents*(j)
+```
+
+That is, we check if `find(uf, i) == find(uf, j)`.
+"""
 struct UnionFind
   parents::Vector{Id}
 end
 
+"""
+    UnionFind()
+
+Creates a new empty unionfind.
+"""
```

Comment:
```suggestion
"""
    UnionFind()

Creates a new empty unionfind.
"""
```

Can you move it to the struct docstring please?